### PR TITLE
Additional logic to up/downscale

### PR DESCRIPTION
**Pull request by dh-harald**
resolves https://github.com/hjacobs/kube-downscaler/issues/47

### CODE REVIEW

**Comment by hjacobs** [comment]
Can you explain the behavior in the README? I think it's getting more and more complex to understand the whole logic, I suggest adding a separate section to explain this new "alternative downscale once" behavior.

File: kube_downscaler/scaler.py [hunk 1]
```diff
@@ -46,11 +48,24 @@ def autoscale_resource(resource: pykube.objects.NamespacedAPIObject,
...
 
             if forced_uptime or (exclude and original_replicas):
                 uptime = "forced"
                 downtime = "ignored"
                 is_uptime = True
+            elif upscale_period != 'never' or downscale_period != 'never':
+                uptime = resource.annotations.get(UPSCALE_PERIOD_ANNOTATION, upscale_period)
+                downtime = resource.annotations.get(DOWNSCALE_PERIOD_ANNOTATION, downscale_period)
+                if helper.matches_time_spec(now, uptime) and helper.matches_time_spec(now, downtime):
+                    logger.debug('Upscale period and downscale period are overlaps, do nothing')
```
Comment (hjacobs) [inline kube_downscaler/scaler.py]: grammar: "Upscale and downscale periods overlap, do nothing"

File: kube_downscaler/scaler.py [hunk 2]
```diff
@@ -46,11 +48,24 @@ def autoscale_resource(resource: pykube.objects.NamespacedAPIObject,
...
         else:
             replicas = resource.replicas
+            ignore = False
 
             if forced_uptime or (exclude and original_replicas):
                 uptime = "forced"
                 downtime = "ignored"
                 is_uptime = True
+            elif upscale_period != 'never' or downscale_period != 'never':
+                uptime = resource.annotations.get(UPSCALE_PERIOD_ANNOTATION, upscale_period)
```
Comment (hjacobs) [inline kube_downscaler/scaler.py]: maybe use a different variable name (as `uptime` is already used in the other if branch)

Comment (dh-harald) [reply kube_downscaler/scaler.py]: I used this variable, because of logging later

**Pull request merged**